tidypolars/tibble.py

```diff
@@ -949,6 +960,110 @@ def write_parquet(self,
         """Write a data frame to a parquet"""
         return super().write_parquet(file, compression = compression, use_pyarrow = use_pyarrow, **kwargs)
 
+    def group_by(self, group, *args, **kwargs):
+        res = TibbleGroupBy(self, group, maintain_order=True)
+        return res
+
+    def nest(self, by, select='all', nested_name="data", *args, **kwargs):
```

Comment:
We should try and follow `tidy::nest()` arguments as much as possible. This seems like it's sort of `tidyr::nest()` but also sort of `dplyr::nest_by()`

(See `?tidyr::nest` if you're looking for why I'm suggesting the below changes.)

If defining `nest()` these need to be the args
```python
def nest(self, _by = None, **kwargs):
```
where
* `**kwargs` are name-selection pairs
* `_by` are the columns to group by

The call follows the way `tidyr` now recommends `nest()` is used with similar argument names

Usage would look like this

```python
df.nest(data = ["a", "b"], _by = "c")
```

Reply:
Again, I don't think using "_" for every function argument in R that starts with dot is a good idea. 

For the signature, it makes sense.

Reply:
Continuing the point from above you could feasibly do something like this because of the underscore

```python
df.nest(by = ["a", "b"], _by = "c)
```

Reply:
I matched the function as close as possible to the tidyr's nest(), but without the "_" for by.